src/components/znkExercise/controllers/baseZnkExercise.controller.js

```diff
@@ -37,6 +34,28 @@
                 return exercise.time;
             }
 
+            function _finishExercise() {
+                exerciseResult.isComplete = true;
+                exerciseResult.endedTime = Date.now();
+                exerciseResult.$save();
+
+                var exerciseTypeValue = ExerciseTypeEnum.getValByEnum(exerciseData.exerciseTypeId);
```

Comment:
why not ExerciseTypeEnum.getValByEnum(exerciseData.exerciseTypeId).tolowercase()?